fastlane_bot/events/exchanges/solidly_v2.py

```diff
@@ -41,31 +49,49 @@ def add_pool(self, pool: Pool):
         self.pools[pool.state["address"]] = pool
 
     def get_abi(self):
-        return VELOCIMETER_V2_POOL_ABI if "velocimeter" in self.exchange_name else SOLIDLY_V2_POOL_ABI
+        try:
```

Comment:
Don't "try".

Use "if key in dict then do stuff, else raise exception".

A "try" can fail for some unexpected reason (in opposed to the one that you're expecting it to fail on), which will make it indistinguishable from the expected reason of a failure in this case.

Reply:
> Don't "try".
> 
> Use "if key in dict then do stuff, else raise exception".
> 
> A "try" can fail for some unexpected reason (in opposed to the one that you're expecting it to fail on), which will make it indistinguishable from the expected reason of a failure in this case.

The approach of using "if-then-else" logic to check for key presence in a dictionary before proceeding, as opposed to employing "try-except" for control flow and error handling, has its own trade-offs. While it addresses the concern of catching unexpected exceptions and makes the failure reasons more explicit, it also has potential downsides compared to the original "try-except" approach:

1. **Performance Considerations:** In scenarios where the expected path is the absence of errors (i.e., the key exists in the dictionary most of the time), the "try-except" pattern can be more efficient. This is because accessing a key directly and catching an exception if it doesn't exist avoids the overhead of checking the condition each time. Python's exception handling is very efficient when exceptions are rare.

2. **Readability and Pythonic Style:** Python's philosophy encourages using exceptions for control flow in certain scenarios, following the EAFP (Easier to Ask for Forgiveness than Permission) principle. This principle suggests that attempting an operation directly and handling exceptions if they occur can lead to cleaner and more readable code than pre-checking conditions. The original approach aligns more with this Pythonic style, making it more idiomatic to Python developers.

3. **Complexity in Handling Multiple Keys:** If you need to access multiple keys or perform several operations that might fail, a "try-except" block can handle all these cases concisely without multiple nested "if" checks. This can keep the code less nested and more straightforward.

4. **Error Specificity:** While the "if-then-else" approach provides clarity on why an operation failed (by checking conditions explicitly), a well-designed "try-except" block can also capture specific exceptions and provide detailed error messages. The distinction becomes less about the ability to provide error specificity and more about the stylistic and performance choices in handling expected vs. unexpected conditions.

Reply:
Performance is negligible (here in particularly).
But TBH, since the `else` case (or the `except` case if using `try`) is used here just for raising an exception, it would probably be cleanest to simply do the desired operation.
In other words - no need for neither if/else nor try/except.
Simply access the dictionary, and any exception thrown as a result of that should be easy to analyze.

Reply:
Updated